File: java/src/main/java/mx/iteso/Approve.java
package mx.iteso;

/**
 * Class to approve permission.
 */
public class Approve extends Privilege {
    /**
     * adds user to test.
     */
    private User user;

    /**
     *
     * @param us adds the user using the permissions.
     */
    public Approve(final User us) {
        this.user = us;
    }

    /**
     *
     *  @return user description.
     */
    public String getDescripcion() {
        return user.getDescripcion() + "can approve";
    }

    /**
     *
     * @return approve permissions, in this case must be true.
     */
    @Override
    public boolean canApprove() {
        return true;
    }

    /**
     *
     * @return comment permissions.
     */
    @Override
    public boolean canComment() {
        return user.canComment();
    }

    /**
     *
     * @return delete permissions.
     */
    @Override
    public boolean canDelete() {
        return user.canComment();
    }


    /**
     *
     * @return publish permissions.
     */
    @Override
    public boolean canPublish() {
        return user.canPublish();
    }

}

File: python/approve.py
from PrivilegeDecorator import PrivilegeDecorator


class Approve(PrivilegeDecorator):
    PRIVILEGE = 'can approve'

    def __init__(self, newUser):
        self.user = newUser

    def getUser(self):
        return self.user

    def delete(self):
        return self.user.delete()

    def approve(self):
        return self.PRIVILEGE

    def publish(self):
        return self.user.publish()

    def comment(self):
        return self.user.comment()
